src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs

```diff
@@ -133,11 +138,29 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                 if (parentActivity == null)
                 {
                     // telemetryContext.Id is always set: if it was null, it is set to opTelemetry.Id and opTelemetry.Id is never null
-                    operationActivity.SetParentId(telemetryContext.Id);
+                    if (Activity.DefaultIdFormat == ActivityIdFormat.W3C)
+                    {
+                        if(W3CUtilities.IsCompatibleW3CTraceID(telemetryContext.Id))
+                        {
+                            operationActivity.SetParentId(ActivityTraceId.CreateFromString(telemetryContext.Id.AsSpan()), ActivitySpanId.CreateRandom());
```

Comment:
don't create random parent span id - use invalid 00000... so we can later know there was no valid parent.